components/server/src/jobs/runner.ts

```diff
@@ -108,7 +108,7 @@ export class JobRunner {
                 }
             });
         } catch (err) {
-            if (err instanceof ResourceLockedError) {
+            if (err instanceof ResourceLockedError || err instanceof ExecutionError) {
```

Comment:
I moved the catching to the job runner, because I want other places to fail if we were not able to execute.